test/Validus.Tests/ValidatorOperationsTests.fs

```diff
@@ -0,0 +1,64 @@
+module Validus.ValidatorOperations.Tests
+
+open Xunit
+open Validus
+open Validus.Operators
+open FsUnit.Xunit
+open FsToolkit.ErrorHandling
+
+let notStartsWithWhiteSpace fieldName (s: string) =
+    if s.StartsWith ' '
+    then Error <| ValidationErrors.create fieldName [ $"%s{fieldName} can't start with whitespace" ]
+    else Ok <| s
+
+let notEndsWithWhiteSpace fieldName (s: string) =
+    if s.EndsWith ' '
+    then Error <| ValidationErrors.create fieldName [ $"%s{fieldName} can't end with whitespace" ]
+    else Ok <| s
+
+let compositionValidator = notStartsWithWhiteSpace <+> notEndsWithWhiteSpace
+
+let chainValidator = Validators.Default.String.notEmpty >=> compositionValidator
```

Comment:
@pimbrouwers here it is

Reply:
Thanks! I saw those. But I guess I still don't see why it's a problem to run all three each time?

Reply:
The null reference emerges because the `notStartsWithWhiteSpace` and `notEndsWithWhiteSpace` are not null-safe. Since this only impacts strings, I don't think we need a setup where these validators act as "null guards".

Reply:
I don't understand what you mean by that

Reply:
The problem isn't in the library, it's the your validator rule is not null-safe. `s.StartsWith ' '` will throw an exception if the value is `null`, which is what you're experiencing here. Not something necessarily wrong within the library.

Reply:
But I don't want it to execute it at all if previous failed

Reply:
You can also wrap these together into your own _single_ validator, which does this successively.

Reply:
See: https://github.com/pimbrouwers/Validus#creating-a-bespoke-validator

Reply:
I don't understand why validators chaining and an operator for that looks unnecessary for you

Reply:
It just creates a branch in the cognitive model, which I am not a fan of. But if you are so insistent, I will.experiment a little with my own solution and well come back to the drawing board.

Reply:
fwiw, apart from the choice of operator, I think that the mental model of `validateX --> andThen --> validateY --> andThen --> validateZ` is fine. The choice of examples here is weird though, I kind think of any reason why you'd have `notStartsWithWhitespace` and `notEndsWithWhitespace`. You either have whitespace on both ends (in which case you can disallow it or strip it), but rarely would you allow it on one end only.

Anyway, that's just my 2p. I think composition is good. But I'm not sure if Kleisli is a good choice here, though the functionality is pretty close to Kleisli, all the functions than should actually be `'a -> M<'a>`.